meson.build

```diff
@@ -33,6 +33,20 @@ add_project_link_arguments(
     language: 'c',
 )
 
+# in release mode, we automatically turn off cast checks and asserts
+if get_option('buildtype').startswith('release')
+    add_project_arguments('-DNDEBUG', language : ['cpp', 'c'])
```

Comment:
https://mesonbuild.com/Builtin-options.html#base-options

You can set `default_options: [''b_ndebug=if-release']` in the `project()` defaults.

Reply:
thanks! I think I've followed your suggestions